### PR TITLE
Dataset ops v2.6

### PR DESCRIPTION
**Pull request by regit**
Update of #7732 addressing the comments.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
- https://redmine.openinfosecfoundation.org/issues/5184
- https://redmine.openinfosecfoundation.org/issues/5383

Describe changes:
- rework IPv6 parsing
- fix size in content matching

suricata-verify-pr: 908

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | test | baseline | % |
|--|--|--|--| 
| | generic_stats_chk | | |
| .capture.kernel_drops | 5007798 | 5654519 | 88.56% |
| .flow.end.state.new | 19998 | 14867 | 134.51% |
| .flow.end.tcp_state.syn_sent | 1471 | 183 | 803.83% |
| .tcp.segment_memcap_drop | 32914 | 11729 | 280.62% |
| .tcp.reassembly_gap | 176670 | 114099 | 154.84% |
| .tcp.insert_data_normal_fail | 31117 | 11358 | 273.97% |
| .app_layer.error.http.parser | 83 | 55 | 150.91% |
| .app_layer.error.smtp.gap | 75 | 61 | 122.95% | 

Pipeline 8707
WARNING: THERE IS A KNOWN BAD BASELINE WITH PACKET DROPS. bE MINDFUL OF ANY RESULTS.

File: doc/userguide/rules/ipaddr.rst [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+------
+
+The `ip.dst` keyword is a sticky buffer to match on destination IP address. It matches on the binary representation
+and is compatible with the dataset of type `ip` and `ipv4`.
+
+Example:
+
+::
+
+ alert tcp $HOME_NET any -> any any (msg:"Outbound bad list"; flow:to_server; ip.dst; dataset:isset,badips,type ip,load badips.list; sid:1; rev:1;)
```
Comment (jufajardini) [inline doc/userguide/rules/ipaddr.rst:31]: Is `flow:to_server` what we really want, here?

**Review by jufajardini — COMMENTED** [review]
Doc changes look good to me! Just one comment - not sure if a change is required or not.

File: doc/userguide/rules/ipaddr.rst (hunk 1 above)
Comment (regit) [reply doc/userguide/rules/ipaddr.rst:31]: Yes (but could be to_server,established): outbound connection from inside to a bad IP.

Comment (jufajardini) [reply doc/userguide/rules/ipaddr.rst:31]: I see! thanks for explaining :)

**Review by jufajardini — APPROVED** [review]
doc changes lgtm :)

**Comment by victorjulien** [comment]
replaced by #8032

**Pull request closed without merging**